### DISCUSSION TITLE
Polling with a NgRx Effect

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#297] [Announcements]
# Polling with a NgRx Effect

A snippet on how to poll a service by using a NgRx Effect.

https://timdeschryver.dev/blog/polling-with-a-ngrx-effect

**Comment by Coconaut5** [comment]
Hello there! I know it was awhile ago this was posted but I thought its worth a short anyways. Im trying to implement a polling effect in NGRX for a project and everything seems to be wroking pretty fine. However i have the issue that every single poll cycle seems to run change detection (maybe its supposed to?) but also update the store state and even the selector even though the response is exactly the same... Which then causes issues in the component with for example expansion panels open state being reset every single cycle.

Im using NGRX Entity library and just doing a regular setAll, hoever I was expecting it to just sort of handle this for me? Do you have any clue on how to solve this. How have you solved it in this solution? 

Many thanks!

**Reply by timdeschryver** [reply]
This is expected. While the selector values are the same, the reference within the state will be modified and thus the selectors will re-execute.
As a solution you can write custom checks instead of using the default reference check (`===`), for an example see [Help Angular to make your application faster](https://timdeschryver.dev/blog/help-angular-to-make-your-application-faster#referential-checks-ngrx)

**Reply by Coconaut5** [reply]
Thank you for the response! I ended up implementing the custom check, however in this case I put it in the reducer. Is there any practices against that? I assumed it would be better if I could avoid potential store updates and selectors triggering right at the source?

**Reply by timdeschryver** [reply]
Yea, that's fine @Coconaut5
